gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/tasks/CreateManifestTask.java

```diff
@@ -98,6 +98,8 @@
 
     private final ListProperty<ProductDependency> productDependencies =
             getProject().getObjects().listProperty(ProductDependency.class);
+    private final SetProperty<ProductId> optionalProductIds =
```

Comment:
Should this be `ProductDependency` instead of `ProductId`? Optional product deps can have version info?

Reply:
Or maybe there are two cases:

1. A product dependency is discovered that you want to make optional
2. You want to manually add an optional product dependency

?

Reply:
Ah, silly me, you could just add the manual optional product dependency like so:

```gradle
productDependencies {
	// ...
	optional = true
}
```

Reply:
So am I right in thinking this is really this is just for discovered optional pdeps?

Reply:
They can if they are defined as explicit product dependencies with an optional field.
They way I see the `optionalProductDependency` here is just as a flag to mark a discovered dependency as optional.

Reply:
I was answering to the first comment above. Yes to the last question, that bit is just for discovered dependencies.